flink-streaming-connectors/flink-connector-hbase/src/main/java/org/apache/flink/streaming/connectors/hbase/MutationActionList.java

```diff
@@ -0,0 +1,371 @@
+		return this;
+	}
+
+	/**
+ 	 * Add to {@link MutationActionList} a {@link MutationAction} of type {@link MutationAction.Type#DELETE_FAMILY},
+	 * which will create an HBase {@link Delete} operation that deletes versions of the specified family with
+	 * timestamps less than or equal to the specified timestamp.
+	 *
+	 * @param family family name
+	 * @param timestamp version timestamp
+	 * @return this
+	 */
+	public MutationActionList addDeleteFamily(byte[] family, long timestamp) {
+		actions.add(new MutationAction(family, null, null, timestamp, 0, MutationAction.Type.DELETE_FAMILY));
+		return this;
+	}
+
+	/**
+	 * Add to {@link MutationActionList} a {@link MutationAction} of type {@link MutationAction.Type#DELETE_FAMILY},
+	 * which will create an HBase {@link Delete} operation that deletes all versions of the specified family.
+	 *
+	 * @param family family name
+	 * @return this
+	 */
+	public MutationActionList addDeleteFamily(byte[] family) {
+		actions.add(new MutationAction(family, null, null, Long.MAX_VALUE, 0, MutationAction.Type.DELETE_FAMILY));
+		return this;
+	}
+
+	/**
+	 * Add to {@link MutationActionList} a {@link MutationAction} of type {@link MutationAction.Type#DELETE_COLUMNS},
+	 * which will create an HBase {@link Delete} operation that deletes versions of the specified column with
+	 * timestamps less than or equal to the specified timestamp.
+	 *
+	 * @param family family name
+	 * @param qualifier column qualifier
+	 * @param timestamp version timestamp
+	 * @return this
+	 */
+	public MutationActionList addDeleteColumns(byte[] family, byte[] qualifier, long timestamp) {
+		actions.add(new MutationAction(family, qualifier, null, timestamp, 0, MutationAction.Type.DELETE_COLUMNS));
+		return this;
+	}
+
+	/**
+	 * Add to {@link MutationActionList} a {@link MutationAction} of type {@link MutationAction.Type#DELETE_COLUMNS},
+	 * which will create an HBase {@link Delete} operation that deletes all versions of the specified column.
+	 *
+	 * @param family family name
+	 * @param qualifier column qualifier
+	 * @return this
+	 */
+	public MutationActionList addDeleteColumns(byte[] family, byte[] qualifier) {
+		actions.add(new MutationAction(family, qualifier, null, Long.MAX_VALUE, 0, MutationAction.Type.DELETE_COLUMNS));
+		return this;
+	}
+
+	/**
+	 * Add to {@link MutationActionList} a {@link MutationAction} of type {@link MutationAction.Type#DELETE_COLUMN},
+	 * which will create an HBase {@link Delete} operation that deletes the specified version of the specified column.
+	 *
+	 * @param family family name
+	 * @param qualifier column qualifier
+	 * @param timestamp version timestamp
+	 * @return this
+	 */
+	public MutationActionList addDeleteColumn(byte[] family, byte[] qualifier, long timestamp) {
+		actions.add(new MutationAction(family, qualifier, null, timestamp, 0, MutationAction.Type.DELETE_COLUMN));
+		return this;
+	}
+
+	static class MutationAction {
```

Comment:
addXXX() APIs can cover all the flavors of MutationActions (including delete entire family, delete all columns and delete the most recent column you mentioned above) and help users create them. I think it would be easier to use than let users create these actions themselves.